functions/Add-DbaAgDatabase.ps1

```diff
@@ -116,6 +116,11 @@ function Add-DbaAgDatabase {
         [switch]$UseLastBackup,
         [switch]$EnableException
     )
+    begin {
+        if (!($SqlInstance -or $InputObject -or $MyInvocation.ExpectingInput)) {
```

Comment:
We generally use `Test-Bound SqlInstance -Not` (returns true if the parameter was not provided) or `Test-Bound SqlInstance` will return false if not provided.

I have not had chance to update contributing guide with these bits...

As well, is there a reason `$MyInvocation.ExpectingInput` is needed?

Reply:
Thanks.  I'll update the code to use that instead.  That likely won't be until later next week due to the holidays.  As to `$MyInvocation.ExpectingInput`, check the Learning section above.

Reply:
Simple fix there is to simply check if these parameters are provided in the process block and not the Begin...that is the pattern used in all our other commands.